x/axelarnet/keeper/msg_server.go

```diff
@@ -407,7 +407,10 @@ func (s msgServer) routeInnerMsg(ctx sdk.Context, msg sdk.Msg) (*sdk.Result, err
 
 // toICS20 converts a cross chain transfer to ICS20 token
 func toICS20(ctx sdk.Context, k types.BaseKeeper, transfer nexus.CrossChainTransfer) sdk.Coin {
-	path, _ := k.GetIBCPath(ctx, transfer.Recipient.Chain.Name)
+	// if chain or path not found, it will create coin with base denom
+	chain, _ := k.GetCosmosChain(ctx, transfer.Asset.GetDenom())
```

Comment:
It seems like we should be verifying that `chain` and `transfer.Recipient.Chain.Name` match somewhere as a validation check?

Reply:
Nvm, I realized that we care about the origin chain of the denom, whereas the recipient chain can be different.

Reply:
yep